composer/loggers/slack_logger.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Log metrics to slack, using Slack postMessage api - https://api.slack.com/methods/chat.postMessage.
+
+First export 2 environment variable to use this logger.
+1. SLACK_LOGGING_API_KEY: To get app credentials, follow tutorial here - https://api.slack.com/tutorials/tracks/posting-messages-with-curl?app_id_from_manifest=A053W1QCEF2
+2. SLACK_LOGGING_CHANNEL_ID: Channel id to send the message (Open slack channel in web browser to look this up)
+
+Next write script to output metrics / hparams / traces to slack channel. See example below:
+
+trainer = Trainer(
+    model=mnist_model(num_classes=10),
+    train_dataloader=train_dataloader,
+    max_duration='2ep',
+    algorithms=[
+        LabelSmoothing(smoothing=0.1),
+        CutMix(alpha=1.0),
+        ChannelsLast(),
+    ],
+    loggers=[
+        SlackLogger(
+            log_metrics_config=SlackLogConfig(
+                formatter_func=(lambda data, **kwargs: [{
+                    'type': 'section',
+                    'text': {
+                        'type': 'mrkdwn',
+                        'text': f'*{k}:* {v}'
+                    }
+                } for k, v in data.items()]),
+                include_keys={'loss/train/total'},
+                interval_in_seconds=1
+            ),
+            log_hparams_config=SlackLogConfig(),
+        )
+    ],
+)
+
+trainer.fit()
+"""
+
+import os
+import time
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple
+
+from composer.loggers.logger_destination import LoggerDestination
+from composer.utils import MissingConditionalImportError
+
+
+class SlackLogConfig:
+    """Log config for slack logger, including formatter function, keys to include, and interval in seconds.
+
+    formatter_func ((...) -> Any | None): A formatter function that returns list of blocks to be sent to slack
+    include_keys (Set[str]): A set of metric/logs/traces keys to include in the message. If None, no keys are included.
```

Comment:
what do you think about defaulting to all keys if `include_keys=None`. I think that behavior might be more useful as a default? not completely sure though

Reply:
yeah good point, None and empty should probably have different behavior, even though there's no use case for an empty set. 

Will make the change and update the docs.